nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,149 @@
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable clight or not.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clight;
+      defaultText = "pkgs.clight";
+      description = ''
+        Clight derivation to use.
+      '';
+    };
+
+    clightd = mkOption {
+      type = types.package;
+      default = pkgs.clightd;
+      defaultText = "pkgs.clightd";
+      description = ''
+        Clightd derivation to use.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Additional configuration to pass to clight.conf.
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = ''
+        Additional command-line arguments to pass to
+        <command>clight</command>.
+      '';
+    };
+
+    latitude = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Your current latitude, between
+        <literal>-90.0</literal> and <literal>90.0</literal>. Must be provided
+        along with longitude.
+      '';
+    };
+
+    longitude = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Your current longitude, between
+        between <literal>-180.0</literal> and <literal>180.0</literal>. Must be
+        provided along with latitude.
+      '';
+    };
+
+    temperature = {
+      day = mkOption {
+        type = types.int;
+        default = 5500;
+        description = ''
+          Colour temperature to use during the day, between
+          <literal>1000</literal> and <literal>25000</literal> K.
+        '';
+      };
+      night = mkOption {
+        type = types.int;
+        default = 3700;
+        description = ''
+          Colour temperature to use at night, between
+          <literal>1000</literal> and <literal>25000</literal> K.
+        '';
+      };
+    };
+  };
+
+  config = let
+    manualLocation = cfg.latitude != null && cfg.longitude != null;
+  in mkIf cfg.enable {
+    boot.kernelModules = [ "i2c_dev" ];
+    environment.systemPackages = [ cfg.clightd cfg.package ];
+    services.dbus.packages = [ cfg.clightd cfg.package ];
+    services.upower.enable = true;
+
+    services.geoclue2 = mkIf (!manualLocation) {
+      enable = true;
+      appConfig."clightc" = {
+        isAllowed = true;
+        isSystem = true;
+      };
+    };
+
+    systemd.services.clightd = {
+      requires = [ "polkit.service" ];
+      wantedBy = [ "multi-user.target" ];
+
+      description = "Bus service to manage various screen related properties (gamma, dpms, backlight)";
+      serviceConfig = {
+        Type = "dbus";
+        BusName = "org.clightd.clightd";
+        Restart = "on-failure";
+        RestartSec = 5;
+      };
+
+      script = ''
+        ${cfg.clightd}/lib/clightd/clightd
```

Comment:
You can use `serviceConfig.ExecStart = "${cfg.clightd}/lib/clightd/clightd"` if it's just a single command. Same with `clight`

Reply:
It's also a bit weird that an executable is in `$out/lib`. It would be cool if you could move it to `$out/bin` as is convention.